src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsNuGetFramework.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>A type that represents the components of a .NET Target Framework Moniker.</summary>
+    /// <remarks><see cref="System.Runtime.Versioning.FrameworkName"/> does not support .NET 5 Target Framework Monikers with a platform, but this type does.</remarks>
+    [ComImport]
+    [Guid("E57318D0-9A4D-443C-87F6-631D7F6B14CF")]
+    public interface IVsNuGetFramework
+    {
+        /// <summary>The framework moniker.</summary>
+        string TargetFrameworkMoniker { get; }
+
+        /// <summary>The platform moniker.</summary>
+        string TargetPlatformMoniker { get; }
+
+        /// <summary>The framework platform version.</summary>
+        string TargetPlatformMinVersion { get; }
```

Comment:
I really wish we could just represent the framework through TPM only :( 
UWP is forcing us to have this abstraction. 

The trick is that the TPMV is used as input, but never as an output.

Reply:
Yep, very sucky. If you have a time machine we could try changing this design. Otherwise I don't have any better ideas 😞

Reply:
NuGetFramework as an input can contain all 3 properties...NuGetFramework as an output can contain only 3.

Reply:
Yes, that's what I meant 🤕

Reply:
The trick is that for UWP, the targetplatform + min version are used. 

If you provide nuget will targetplatform + targetplatforminversion, we say UAP is a target framework identifier, not a platform one.